src/components/generator/generatorFunction.js

```diff
@@ -1,131 +1,147 @@
-
-    // this adds a heuristics which chooses tricks depending on how it fits towards the expected avgDifficulty,
-    // but the difficultyLevels are modified with a random value of a Normal Distribution (which is depending on how extrem the avgDifficulty is),
-    // to have some additional randomness and not get always the same combo
-    if (avgDifficulty != -1) {
-      for (let j = 0; j < possibleTricks.length; j++) {
-        possibleTricks[j].randomDiff = possibleTricks[j].difficultyLevel + randn_bm(-maxDifficulty + 2*Math.abs(maxDifficulty/2 - avgDifficulty), maxDifficulty - 2*Math.abs(maxDifficulty/2 - avgDifficulty));
-      }
-      possibleTricks.sort(
-        (a,b) => (
-          // heuristics for finding a combo with certain avrDifficultie
-          Math.abs(avgDifficulty - (sumOfDifficulties + b.randomDiff)/(i+1)) - Math.abs(avgDifficulty - (sumOfDifficulties + a.randomDiff)/(i+1))
-        )
-      );
-      randomTricks[i] = possibleTricks[possibleTricks.length - 1];
-      sumOfDifficulties += randomTricks[i].difficultyLevel;
-    }
-    else {
-      // take an random trick
-      randomTricks[i] = possibleTricks[Math.floor(Math.random()*possibleTricks.length)];
-    }
+  return true
+}
+
+/**
+ * Find the first trick depending on whether it needs to be from a certain starting position or not.
+ */
+function firstTrick(availableTricks, isStartPositionFixed, startPosition) {
+  if (!isStartPositionFixed) {
+    return availableTricks[randomIndex(availableTricks.length)]
+  }
+  let possibleTricks = availableTricks.filter(trick => trick.startPos === startPosition)
+  if (possibleTricks.length === 0) {
+    throw "The selected options do not contain any trick with the specified starting position."
   }
+  return possibleTricks[randomIndex(possibleTricks.length)]
+}
 
-  //check integrety of combo
-  for (let i = 1; i < randomTricks.length; i++) {
-    let prev = randomTricks[i - 1];
-    let trick = randomTricks[i];
-    if (prev.endPos !== trick.startPos && !arePositionsSimilar(trick.startPos, prev.endPos) && !allowTransitions) {
-      return alert("trick generator is broken");
-    }
+/**
+ * This adds a heuristics which chooses tricks depending on how it fits towards the expected avgDifficulty,
+ * but the difficultyLevels are modified with a random value of a Normal Distribution (which is depending on how
+ * extreme the avgDifficulty is), to have some additional randomness and not get always the same combo.
+ * Original Author: weberax
+ */
+function randomTrickOfAvgDifficulty(possibleTricks, maxDifficulty, avgDifficulty, sumOfDifficulties, i) {
+  for (let j = 0; j < possibleTricks.length; j++) {
+    let min = -maxDifficulty + 2 * Math.abs(maxDifficulty / 2 - avgDifficulty)
+    let max = maxDifficulty - 2 * Math.abs(maxDifficulty / 2 - avgDifficulty)
+    possibleTricks[j].randomDiff = possibleTricks[j].difficultyLevel + randomNormalDist(min, max);
```

Comment:
I am not sure what exactly this does. But it might make sense to extract the logic inside the loop into a separate function, e.g.
```js
calculateRandomDiffForTrick(possibleTricks[j])
```

Also.. what exactly is Diff in this context?

Reply:
Approved